addons/binding/org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/RussoundHandlerFactory.java

```diff
@@ -50,25 +51,34 @@ protected ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
         if (thingTypeUID.equals(RioConstants.BRIDGE_TYPE_RIO)) {
-            return new RioSystemHandler((Bridge) thing);
+            final RioSystemHandler sysHandler = new RioSystemHandler((Bridge) thing);
+            registerThingDiscovery(sysHandler);
+            return sysHandler;
         } else if (thingTypeUID.equals(RioConstants.BRIDGE_TYPE_CONTROLLER)) {
             return new RioControllerHandler((Bridge) thing);
-        } else if (thingTypeUID.equals(RioConstants.BRIDGE_TYPE_SOURCE)) {
-            return new RioSourceHandler((Bridge) thing);
-        } else if (thingTypeUID.equals(RioConstants.BRIDGE_TYPE_ZONE)) {
-            return new RioZoneHandler((Bridge) thing);
-        } else if (thingTypeUID.equals(RioConstants.BRIDGE_TYPE_BANK)) {
-            return new RioBankHandler((Bridge) thing);
-        } else if (thingTypeUID.equals(RioConstants.THING_TYPE_BANK_PRESET)) {
-            return new RioPresetHandler(thing);
-        } else if (thingTypeUID.equals(RioConstants.THING_TYPE_ZONE_PRESET)) {
-            return new RioPresetHandler(thing);
-        } else if (thingTypeUID.equals(RioConstants.THING_TYPE_SYSTEM_FAVORITE)) {
-            return new RioFavoriteHandler(thing);
-        } else if (thingTypeUID.equals(RioConstants.THING_TYPE_ZONE_FAVORITE)) {
-            return new RioFavoriteHandler(thing);
+        } else if (thingTypeUID.equals(RioConstants.THING_TYPE_SOURCE)) {
+            return new RioSourceHandler(thing);
+        } else if (thingTypeUID.equals(RioConstants.THING_TYPE_ZONE)) {
+            return new RioZoneHandler(thing);
         }
 
         return null;
     }
+
+    /**
+     * Registers a {@link RioSystemDeviceDiscoveryService} from the passed {@link RioSystemHandler} and activates it.
+     *
+     * @param bridgeHandler the {@link RioSystemHandler} for discovery services
+     */
+    private synchronized void registerThingDiscovery(RioSystemHandler bridgeHandler) {
+        RioSystemDeviceDiscoveryService discoveryService = new RioSystemDeviceDiscoveryService(bridgeHandler);
+        logger.trace("Try to register Discovery service on BundleID: {} Service: {}",
+                bundleContext.getBundle().getBundleId(), DiscoveryService.class.getName());
+
+        Hashtable<String, String> prop = new Hashtable<String, String>();
```

Comment:
Unfortunately the bundleContext.registerService call requires a Dictionary<String, ?> - of which Hashtable is the only Dictionary implementation (there might be something in apache utils that I'm unaware of).  Ignoring this one